test/test_helper.rb

```diff
@@ -102,13 +102,14 @@ def reset_logger
       end
     end
 
-    def assert_deploy_failure(result)
+    def assert_deploy_failure(result, cause = nil)
       if ENV["PRINT_LOGS"]
         assert_equal false, result, "Deploy succeeded when it was expected to fail"
         return
       end
 
       logging_assertion do |logs|
+        assert_match(/failed due to timeouts./, logs) if cause == :timeout
```

Comment:
This should be asserting on `Result: TIMED OUT` (which isn't happening yet in this PR), like L114